### PR TITLE
Make cached and prefetched state available in partial responses.

### LINKED ISSUE #333: Make cached and prefetched state available in partial responses.
Currently, the cached and prefetched state of a response is available to event handlers and callbacks for the `spfdone` event in the `timing` property:

```js
// Check if the response was cached.
evt.detail.response.timing.spfCached;
// Check if the response was prefetched.
evt.detail.repsonse.timing.spfPrefetched
```

However, this information can be useful when processing multipart responses, so make it available in the `timing` object for parts as well.

### PR DESCRIPTION
**Pull request by hanqingl**
Closes #333.

### CODE REVIEW

File: src/client/nav/request.js [hunk 1]
```diff
@@ -219,6 +219,11 @@ spf.nav.request.handleResponseFromCache_ = function(url, options, timing,
   if (options.onPart && response['type'] == 'multipart') {
     var parts = response['parts'];
     spf.array.each(parts, function(part) {
+      if (!part['timing']) {
+        part['timing'] = {}
+      }
+      part['timing']['spfCached'] = timing['spfCached'];
```
Comment (nicksay) [inline src/client/nav/request.js]: Please make these definitive booleans:  `= !!timing['spfCached'];`

**Comment by nicksay** [comment]
@DavidCPhillips @awbraunstein Do you think it makes more sense to set these values to false for the uncached case?  Or omit them?

**Comment by DavidCPhillips** [comment]
If it's known, I always prefer an explicit false.

**Comment by nicksay** [comment]
@hanqingl Can you add explicit false values to the code path(s) that call `onPart` via the XHR?

**Pull request merged**